tests/core/test_state_store.py

```diff
@@ -0,0 +1,63 @@
+# Standard Library
+import os
+import shutil
+import time
+
+# First Party
+from smdebug.core.state_store import StateStore
+
+
+def setup_test1():
+    try:
+        shutil.rmtree("checkpoints_test_dir/")
+    except:
+        pass
+    os.mkdir("checkpoints_test_dir/")
+    os.mkdir("checkpoints_test_dir/subdir1")
+    os.mkdir("checkpoints_test_dir/subdir2")
+    dir_path = os.path.abspath("checkpoints_test_dir")
+    with open(dir_path + "/metadata.json", "w") as f:
+        pass
+    with open(dir_path + "/config_test.json", "w") as f:
+        pass
+    with open(dir_path + "/subdir1/checkpoint_test1.txt", "w") as f:
+        pass
+    with open(dir_path + "/subdir2/checkpoint_test2.txt", "w") as f:
+        pass
+    return dir_path
+
+
+def setup_test2():
+    try:
+        shutil.rmtree("checkpoints_test_dir/")
+    except:
+        pass
+    os.mkdir("checkpoints_test_dir/")
+    dir_path = os.path.abspath("checkpoints_test_dir")
+    with open(dir_path + "/metadata.json", "w") as f:
+        pass
+    return dir_path
+
+
+def test_is_checkpoint_updated():
+    s1 = StateStore()
+    s1._checkpoint_update_timestamp = time.time()
+    s1._checkpoint_dir = setup_test1()
+    # the checkpoint update time is greater than _checkpoint_update_timestamp. is_checkpoint_updated should return true.
+    assert s1.is_checkpoint_updated()
+
+    s1._checkpoint_update_timestamp = time.time()
```

Comment:
why this i set in test case. 

I think you require test case like:
test1:

State_store - check_updated() ?  returns False
create dir/files 
State_store - check_updated() ?  returns True 
update_state() 
state_store -> check_updated ? reutrns False 

update one of existing file : 
state_store -> check_updated ? returns True 
update_state()
Create _one_ new file 
state_store -> check_updated ? returns True